modules/imap/functions.php

```diff
@@ -1217,3 +1217,23 @@ function get_request_params($request) {
 
     return [$server_id, $uid, $folder, $msg_id];
 }}
+
+/**
+ * @subpackage imap/functions
+ */
+if (!hm_exists('parse_sieve_config_host')) {
+function parse_sieve_config_host($host) {
+    $url = parse_url($host);
+    if (! isset($url['host']) || ! isset($url['scheme'])) {
```

Comment:
Does this work when $host is IP address or simple hostname without scheme? It should.

Reply:
@kroky with ip address it works but without scheme it is much difficult to validate if it is a valid url

Reply:
@kroky I also fixed issue when Hm_Sieve_Client_Factory::init returns null

Reply:
When user supplied host = 192.168.0.1 and port = something - this check will fail as parse_url returns only path for IP addresses. I don't think users supply scheme when they supply IP addresses - looks weird to me. Maybe prepend tcp:// scheme if there is no scheme?

Reply:
The host and the port are both provided in a single string. Then I can prepend tcp:// if there is no scheme.

Reply:
Passing the protocol to socket_connect is leading to an. But I noticed that tls option of Client::connect in sieve manager is not used. I will fix this.

Reply:
@josaphatim is this ready now?

Reply:
@kroky I think only tls will be used. Not ssl

Reply:
I mean this https://github.com/cypht-org/php-sieve-manager/blob/master/src/ManageSieve/Client.php#L551 still doesn't use the passed tls parameter. Why bother set and pass it then?